### PR TITLE
Google Identity Resource

### PR DESCRIPTION
**Pull request by nmonterroso**
@Wikia/services-team 
https://wikia-inc.atlassian.net/browse/SERVICES-987

Provide a resource `GET /google/identity` which validates a google `access_token` and returns the resulting `AuthenticatedAccount` (userId, email).

### CODE REVIEW

File: service/external-auth/src/main/java/com/wikia/services/externalauth/gateway/GoogleGateway.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public GoogleGateway(@Named(GOOGLE_CLIENT_ID_KEY) String clientID) {
       LOGGER.error(GOOGLE_API_UNAVAILABLE_MSG, e);
       throw new InternalServerErrorException();
     } catch (Exception e) {
```
Comment (WikiaTech) [inline service/external-auth/src/main/java/com/wikia/services/externalauth/gateway/GoogleGateway.java]: MAJOR Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: service/external-auth/src/test/java/com/wikia/services/externalauth/helpers/TestHelper.java [hunk 2]
```diff
@@ -1,12 +1,14 @@
 package com.wikia.services.externalauth.helpers;
 
 import com.wikia.gateway.helios.domain.AccessData;
+import com.wikia.services.externalauth.domain.AuthenticatedAccount;
 import com.wikia.services.externalauth.domain.LinkedGoogleAccount;
 
 public class TestHelper {
```
Comment (WikiaTech) [inline service/external-auth/src/test/java/com/wikia/services/externalauth/helpers/TestHelper.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1118)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

File: service/external-auth/src/test/java/com/wikia/services/externalauth/helpers/TestHelper.java [hunk 3]
```diff
@@ -22,10 +24,16 @@
...
   public static final int TOKEN_EXPIRES_IN = 123;
 
+  private TestHelper() {
+
+  }
 
   public static final AccessData ACCESS_DATA =
       new AccessData(ACCESS_TOKEN, TOKEN_EXPIRES_IN, VALID_WIKIA_USER_ID);
 
   public static final LinkedGoogleAccount LINKED_GOOGLE_ACCOUNT =
```
Comment (WikiaTech) [inline service/external-auth/src/test/java/com/wikia/services/externalauth/helpers/TestHelper.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)

File: service/external-auth/src/test/java/com/wikia/services/externalauth/helpers/TestHelper.java [hunk 4]
```diff
@@ -22,10 +24,16 @@
...
       new LinkedGoogleAccount(VALID_WIKIA_USER_ID, VALID_GOOGLE_USER_ID);
+
+  public static final AuthenticatedAccount GOOGLE_AUTHENTICATED_ACCOUNT =
```
Comment (WikiaTech) [inline service/external-auth/src/test/java/com/wikia/services/externalauth/helpers/TestHelper.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)

File: service/external-auth/src/test/java/com/wikia/services/externalauth/helpers/TestHelper.java [hunk 5]
```diff
@@ -22,10 +24,16 @@
   public static final String ACCESS_TOKEN = "xYzaBc123";
   public static final int TOKEN_EXPIRES_IN = 123;
 
+  private TestHelper() {
+
+  }
 
   public static final AccessData ACCESS_DATA =
```
Comment (WikiaTech) [inline service/external-auth/src/test/java/com/wikia/services/externalauth/helpers/TestHelper.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 1 major
* MINOR 3 minor

Watch the comments in this conversation to review them.

File: service/external-auth/src/main/java/com/wikia/services/externalauth/gateway/GoogleGateway.java [hunk 6]
```diff
@@ -51,7 +51,11 @@ public GoogleGateway(@Named(GOOGLE_CLIENT_ID_KEY) String clientID) {
...
```
Comment (WikiaTech) [inline service/external-auth/src/main/java/com/wikia/services/externalauth/gateway/GoogleGateway.java]: MAJOR Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: service/external-auth/src/main/java/com/wikia/services/externalauth/gateway/GoogleGateway.java [hunk 7]
```diff
@@ -51,7 +51,11 @@ public GoogleGateway(@Named(GOOGLE_CLIENT_ID_KEY) String clientID) {
...
-      LOGGER.error(GOOGLE_ACCESS_TOKEN_VALIDATION_FAILED_MSG, e);
+      String message = String.format(
```
Comment (sqreek) [inline service/external-auth/src/main/java/com/wikia/services/externalauth/gateway/GoogleGateway.java]: We should start using Markers instead string.format.
It's harded to filter that in kibana

**Comment by sqreek** [comment]
LGTM, but can you change tests method name into format: whenPassingCorrectTokenShouldReturnOk ?

File: service/external-auth/src/main/java/com/wikia/services/externalauth/gateway/GoogleGateway.java (hunk 7 above)
Comment (nmonterroso) [inline service/external-auth/src/main/java/com/wikia/services/externalauth/gateway/GoogleGateway.java]: Good call d9d0c3c

File: service/external-auth/src/main/java/com/wikia/services/externalauth/gateway/GoogleGateway.java [hunk 8]
```diff
@@ -51,7 +54,12 @@ public GoogleGateway(@Named(GOOGLE_CLIENT_ID_KEY) String clientID) {
       LOGGER.error(GOOGLE_API_UNAVAILABLE_MSG, e);
       throw new InternalServerErrorException();
     } catch (Exception e) {
```
Comment (WikiaTech) [inline service/external-auth/src/main/java/com/wikia/services/externalauth/gateway/GoogleGateway.java]: MAJOR Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: service/external-auth/src/main/java/com/wikia/services/externalauth/gateway/GoogleGateway.java [hunk 9]
```diff
@@ -10,6 +11,8 @@
 import com.google.api.client.json.jackson2.JacksonFactory;
 import com.google.api.services.oauth2.Oauth2;
 import com.google.api.services.oauth2.model.Tokeninfo;
+import com.google.common.collect.ImmutableBiMap;
```
Comment (WikiaTech) [inline service/external-auth/src/main/java/com/wikia/services/externalauth/gateway/GoogleGateway.java]: MINOR Remove this unused import 'com.google.common.collect.ImmutableBiMap'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Pull request merged**